pkg/registry/apis/folders/sub_access.go

```diff
@@ -0,0 +1,72 @@
+package folders
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apiserver/pkg/registry/rest"
+
+	"github.com/grafana/grafana/pkg/apis/folder/v0alpha1"
+	"github.com/grafana/grafana/pkg/infra/appcontext"
+	"github.com/grafana/grafana/pkg/services/apiserver/endpoints/request"
+	"github.com/grafana/grafana/pkg/services/folder"
+	"github.com/grafana/grafana/pkg/services/guardian"
+)
+
+type subAccessREST struct {
+	service folder.Service
+}
+
+var _ = rest.Connecter(&subAccessREST{})
+
+func (r *subAccessREST) New() runtime.Object {
+	return &v0alpha1.FolderAccessInfo{}
+}
+
+func (r *subAccessREST) Destroy() {
+}
+
+func (r *subAccessREST) ConnectMethods() []string {
+	return []string{"GET"}
+}
+
+func (r *subAccessREST) NewConnectOptions() (runtime.Object, bool, string) {
+	return nil, false, "" // true means you can use the trailing path as a variable
+}
+
+func (r *subAccessREST) Connect(ctx context.Context, name string, opts runtime.Object, responder rest.Responder) (http.Handler, error) {
+	ns, err := request.NamespaceInfoFrom(ctx, true)
+	if err != nil {
+		return nil, err
+	}
+	user, err := appcontext.User(ctx)
+	if err != nil {
+		return nil, err
+	}
+	f, err := r.service.Get(ctx, &folder.GetFolderQuery{
+		UID:          &name,
+		SignedInUser: user,
+	})
+	if err != nil {
+		return nil, err
+	}
+	guardian, err := guardian.NewByFolder(ctx, f, ns.OrgID, user)
+	if err != nil {
+		return nil, err
+	}
+	canView, err := guardian.CanView()
+	if err != nil || !canView {
+		return nil, fmt.Errorf("not allowed to view")
+	}
```

Comment:
currently we don't need this given the folder service `Get()` [checks the access to the folder](https://github.com/grafana/grafana/blob/ec5bc7c4ab5396ee68f8851de50957a8f578e783/pkg/services/folder/folderimpl/folder.go#L204)
however, probably in the future we will